### ISSUE TITLE
Question: Is it possible to set a menu view within a tab view?

### ISSUE THREAD

**Issue by changalberto** [driftyco/ionic2#405]
I have menu component as the app root, and the menu root page is a tabs component.
I was wondering if is possible to the menu pages as a tab view. This was possible with v1.1

```html
<ion-side-menus>
  <ion-side-menu-content>
    <ion-nav-view></ion-nav-view>
    <ion-tabs class="tabs-positive tabs-icon-only">

      <ion-tab title="Home" icon-on="ion-ios-filing" icon-off="ion-ios-filing-outline">
        <!-- Tab 1 content -->
      </ion-tab>

      <ion-tab title="About" icon-on="ion-ios-clock" icon-off="ion-ios-clock-outline">
        <!-- Tab 2 content -->
      </ion-tab>

      <ion-tab title="Settings" icon-on="ion-ios-gear" icon-off="ion-ios-gear-outline">
        <!-- Tab 3 content -->
      </ion-tab>

    </ion-tabs>
  </ion-side-menu-content>

  <!-- Right menu -->
  <ion-side-menu side="right">
  </ion-side-menu>
</ion-side-menus>
```

Side Menu Pages would simply be shown within
```html
<ion-nav-view></ion-nav-view>
```

Or what would be the recommended approach to do this in version 2?

**Comment by adamdbradley** [comment]
Sure, with v2 `<ion-menu>` no longer wraps their content, but you instead assign the menu which element is the content. Here's an example how using a menu with tabs:
https://github.com/driftyco/ionic2/blob/master/ionic/components/tabs/test/basic/index.ts#L65

Also, you can have multiple menus available, and only enable one at a time:
https://github.com/driftyco/ionic2/blob/master/ionic/components/menu/test/multiple/index.ts

**Comment by changalberto** [comment]
@adamdbradley, with Tabs example you provided, how could I get my side menu items handler set the content?

I'm trying something like this, this isn't working, but I hope you would get what I'm trying to do.

```javascript
  openPage(page) {
    // close the menu when clicking a link from the menu
    this.app.getComponent('menu').close();
    // navigate to the new page if it is not the current page
    // let nav = this.app.getComponent('nav');
    // let menu = this.app.getComponent('menu');
    let tabs = this.app.getComponent('tabs');

    // menu.setContent(page.component);
    // menu.setRoot(page.component);
    // console.log(menu, tabs);
    tabs.setRoot(page.component);
    // nav.setRoot(page.component);
  }
```

Your help much appreciated.

Thanks!

**Comment by adamdbradley** [comment]
Can you set it on `<ion-tabs>`? 
https://github.com/driftyco/ionic2/blob/master/ionic/components/tabs/test/basic/index.ts#L78

**Comment by changalberto** [comment]
Well, I have a my tabs and a side menu.

```html
<ion-menu [content]="content" side="right">

  <ion-toolbar>
    <ion-title>Account</ion-title>
  </ion-toolbar>

  <ion-content>
    <ion-list>
      <button ion-item *ng-for="#p of pages" (click)="openPage(p)">
        {{p.title}}
      </button>
    </ion-list>
  </ion-content>

</ion-menu>

<ion-tabs #content>
  <ion-tab tab-icon="home" tab-title="Home" [root]="dashboard"></ion-tab>
  <ion-tab tab-icon="pin" tab-title="Around Me" [root]="location"></ion-tab>
  <ion-tab tab-icon="chatbubbles" tab-title="Chats" [root]="chats"></ion-tab>
  <ion-tab tab-icon="people" tab-title="Friends" [root]="friends"></ion-tab>
</ion-tabs>
```

I want my menu items content to be treated as if they were tabs. But I do not want it to be a part of the Tab Bar, if this makes any sense.

**Comment by changalberto** [comment]
@adamdbradley sorry to bug you, I know you're swamped with tons of other tickets. I'm working on an app, and I'm sort of stuck at this point. I'm trying to see if there's an ionic way of handling this scenario, or should I do some sort of custom thing with angular2.

Thanks!

**Comment by adamdbradley** [comment]
Not sure I'm fully understanding, but I think you don't want to use `ion-tabs` but instead `ion-nav`. I think what you're looking for is using `setRoot`, like this example: 
https://github.com/driftyco/ionic2-starter/blob/master/www/app/app.html

**Comment by changalberto** [comment]
@adamdbradley 
I was actually using ``ion-nav`` before, and it wasn't working out. But after you made the recommendation  of using ``<ion-tabs #content>`` I'm further along what I'm intending to do. I'm sorry if this is all sounding too confusing, it's a bit hard to articulate without visuals for me. But let try again.

1. Home/Default screen has four tabs. These are top level pages that users can toggle thru.
2. I have four side menu items that are also treated as top level pages.

My problem previously while using ``ion-nav` is that whenever I ``setRoot`` with a side menu page, the ``ion-tabs`` would also be replaced along with it.

So my actual question is:
How do I make the side menu items to toggle content as if they were part of the ``ion-tabs`` instance?

**Comment by mlynch** [comment]
@changalberto the question is then, which tab do you want to load those side menu pages into? You have four, so it's a big ambiguous. Do you have a demo or an app you are trying to emulate?

With our design approach, you *should* be able to do what you want to do, but we just need to understand what it is better so we can give you some code.

**Comment by changalberto** [comment]
@mlynch Alright, I think you're understanding what I'm trying to do now. So to your question: "which tab do you want to load those side menu pages into?" It more like, these menu pages are its own tabs that are part of the ``<ion-tabs>``, but don't appear in the ``tab-nav``.

I have a demo of it using v1.1 in Ionic.io preview. But unfortunately I don't have a way of sharing this with you outside of that.

**Comment by mlynch** [comment]
Okay, I understand. It's like having a page without a corresponding tab icon, is that correct? The side menu is then driving this tab bar programatically?

**Comment by changalberto** [comment]
Yes, Correct! =]

**Comment by mlynch** [comment]
Last question: would the tab bar show on those extra pages, or no?

**Comment by changalberto** [comment]
Yes, the tab bar need to be visible to the users while they're inside a side menu page. This allows the user to navigate back out to the main/core features on the tab quickly without having to navigate backwards.

**Comment by mlynch** [comment]
Okay, so I think the best approach in this case is to have the side menu get the active navigation controller for the active tab and then call setRoot on it. That would keep the tabs the same but navigation would update the tab. Does that seem like an okay solution? Next step would be coming up with code to do that...

**Comment by changalberto** [comment]
Alright, so to dumb it down a little bit to a tutorial form. That means that I would need to go back to having:
```html
<ion-menu [content]="content">

  <ion-toolbar>
    <ion-title>Pages</ion-title>
  </ion-toolbar>

  <ion-content>
    <ion-list>
      <button ion-item *ng-for="#p of pages" (click)="openPage(p)">
        {{p.title}}
      </button>
    </ion-list>
  </ion-content>

</ion-menu>

<ion-nav id="nav" [root]="rootPage" #content swipe-back-enabled="false"></ion-nav>
```

**Comment by mlynch** [comment]
no your example from above with the tabs would be fine, we just need to figure out how to access the nav for the [root] of the active tab, and call setRoot on `that`. That way the tabs stay showing. I need to think through how our API might expose that...

**Comment by changalberto** [comment]
Ok, yes, so I was in the right direction, but I don't think the ``tabs`` component exposes a method where I can do some sort of ``setContent`` or ``setRoot``.

**Comment by changalberto** [comment]
@mlynch Btw, I really appreciate you taking your time to helping me out with this. I'm sort of learning Angular2 and Ionic2 simultaneously, and it's a lot to wrap my head around at the moment. So I appreciate your patience.

**Comment by changalberto** [comment]
@mlynch Any luck with a clean solution on this for the next alpha release?

**Comment by mlynch** [comment]
Hey @changalberto, thanks for hanging in there. Check out this demo for an example of how to do this: https://github.com/driftyco/ionic2/blob/master/ionic/components/tabs/test/ghost/index.ts

Note, you'll need to fetch the latest version of ionic to have the tab reset back to the original root when tapping it, but the "ghost" tab functionality is there. 

Let me know if you have any questions!

**Comment by changalberto** [comment]
@mlynch The ghost tab is working great! But I still can't get it to reset back to original tab using alpha 31.

**Comment by changalberto** [comment]
```typescript
import {ContentChild, QueryList, ViewChildren} from 'angular2/angular2';
import {App, IonicApp, Platform, NavController, Tab} from 'ionic/ionic';

import {HelloIonicPage} from './hello-ionic/hello-ionic';
import {DashboardPage} from './dashboard/dashboard';
import {ListPage} from './list/list';


@App({
  templateUrl: 'app/app.html'
})

class App {
  @ViewChildren(Tab) tab : QueryList<Tab>;

  constructor(app: IonicApp, platform: Platform) {

    // set up our app
    this.app = app;
    this.platform = platform;
    this.initializeApp();

    // set our app's pages
    this.pages = [
      { title: 'Hello Ionic', component: HelloIonicPage },
      { title: 'My First List', component: ListPage }
    ];

    // make HelloIonicPage the root (or first) page
    this.dashboard = DashboardPage;
    this.location = HelloIonicPage;
    this.chats = DashboardPage;
    this.friends = DashboardPage;
  }

  initializeApp() {
    this.platform.ready().then(() => {
      console.log('Platform ready');

      if (typeof StatusBar !== 'undefined') {
        StatusBar.styleDefault();
      }
    });
  }

  openPage(page) {
    // close the menu when clicking a link from the menu
    this.app.getComponent('menu').close();

    this.tab.first.setRoot(page.component);
  }
}
```

@mlynch This is what I got setup so far as test. Let me know if I'm doing something wrong with this setup...

**Comment by mlynch** [comment]
The tab to root is in master only hasn't been released, but it'll work fine here once we release that next week or you can try master

**Comment by changalberto** [comment]
@mlynch Thanks!

**Comment by changalberto** [comment]
@mlynch Also, just to reiterate, Tab active state should be deactivated when selecting menu item, and menu item should be active in place. Thanks!

**Comment by changalberto** [comment]
@mlynch How can you created sub-tabs like in 1.1

![image](https://cloud.githubusercontent.com/assets/24946/11017823/9f5689a8-857a-11e5-9129-c40e83c0386a.png)

**Comment by changalberto** [comment]
@adamdbradley What is the equivalent of [button-bar](http://ionicframework.com/docs/components/#button-bar) in Ionic v2?

**Comment by changalberto** [comment]
@mlynch Can you guys update document on how to set the ghost page as root?

http://ionicframework.com/docs/v2/api/components/tabs/Tabs/
